src/core/IPosisProcess.d.ts

```diff
@@ -3,6 +3,8 @@ interface IPosisProcess {
     imageName: string; // image name (maps to constructor)
     id: PosisPID; // ID
     parentId: PosisPID; // Parent ID
-    log: IPosisLogger; // Logger 
+    log: IPosisLogger; // Logger
+    // For querying extension interfaces (instead of tying ourselves to "levels")
+    queryPosisInterface<T extends keyof PosisInterfaces>(interfaceId: T): PosisInterfaces[T] | undefined;
```

Comment:
As a quick test, I can't figure out howto call queryPosisInterface, also, how is that call going to work in vanilla JS? 
```typescript

type PosisInterfaces = {
  baseKernel: IPosisKernel;
  spawn: IPosisSpawnExtension;
}

let kernel: IPosisKernel = new BaseKernel();

function queryPosisInterface<T extends keyof PosisInterfaces>(interfaceId: T): PosisInterfaces[T] | undefined {
  return kernel;
}

let testkernel: IPosisKernel = queryPosisInterface<IPosisKernel>("baseKernel");
```
`src/index.ts(35,52): error TS2344: Type 'IPosisKernel' does not satisfy the constraint 'PosisInterface'.`